homeassistant/components/insteon/api/config.py

```diff
@@ -0,0 +1,277 @@
+"""API calls to manage Insteon configuration changes."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyinsteon import async_close, async_connect, devices
+from pyinsteon.address import Address
+import voluptuous as vol
+import voluptuous_serialize
+
+from homeassistant.components import websocket_api
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, CONF_DEVICE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from ..const import (
+    CONF_HOUSECODE,
+    CONF_OVERRIDE,
+    CONF_UNITCODE,
+    CONF_X10,
+    DEVICE_ADDRESS,
+    DOMAIN,
+    ID,
+    SIGNAL_ADD_DEVICE_OVERRIDE,
+    SIGNAL_ADD_X10_DEVICE,
+    SIGNAL_REMOVE_DEVICE_OVERRIDE,
+    TYPE,
+)
+from ..schemas import (
+    build_device_override_schema,
+    build_hub_schema,
+    build_plm_manual_schema,
+    build_plm_schema,
+)
+from ..utils import async_get_usb_ports
+
+HUB_V1_SCHEMA = build_hub_schema(hub_version=1)
+HUB_V2_SCHEMA = build_hub_schema(hub_version=2)
+PLM_SCHEMA = build_plm_manual_schema()
+DEVICE_OVERRIDE_SCHEMA = build_device_override_schema()
+OVERRIDE = "override"
+
+
+def get_insteon_config_entry(hass: HomeAssistant) -> ConfigEntry:
+    """Return the Insteon configuration entry."""
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        return entry
+    raise ValueError("No Insteon configuration exists")
+
+
+def add_x10_device(hass: HomeAssistant, x10_device: dict):
+    """Add an X10 device to the Insteon integration."""
+    config_entry = get_insteon_config_entry(hass)
+    options_config = {**config_entry.options}
+    x10_config = config_entry.options.get(CONF_X10, [])
+
+    # Check that we are not adding an existing device
+    for curr_device in x10_config:
+        if (
+            curr_device[CONF_HOUSECODE] == x10_device[CONF_HOUSECODE]
+            and curr_device[CONF_UNITCODE] == x10_device[CONF_UNITCODE]
+        ):
+            raise ValueError("Duplicate X10 device")
```

Comment:
Use `any` instead of the loop:
```py
        if any(
            device[CONF_HOUSECODE] == x10_device[CONF_HOUSECODE]
            and device[CONF_UNITCODE] == x10_device[CONF_UNITCODE]
            for device in x10_config
        ):
            raise ValueError("Duplicate X10 device")
```